src/migration/utilities.js

```diff
@@ -1,14 +1,32 @@
-function getDependency(artifacts, network, deployer, path) {
+function initializeContract(path, deployer, accounts) {
+  const contract = require("truffle-contract")
```

Comment:
I think we might need to add `truffle-contract` to the dependencies or peerDependencies? That might be a "heavy" dependency that we were trying to avoid. @anxolin what do you think?

Reply:
Yeah, I tried to avoid it and put it all just in the npm package.json of 
https://github.com/gnosis/dex-contracts-integration-example
While this is not "very clean", it keeps the size down of dex-contracts npm package.

For which applications is the small size important?

Reply:
Yes, agree with Nick. 

@josojo , super nice that you are doing an example on how to deploy the contracts from another project.

I expect that many projects won't even do this, and they will just use an interface or the original contract and use the rinkeby/mainnet deployments. Sometimes that approach is more straightforward.

Alex, for https://github.com/gnosis/dex-contracts-integration-example I think it would be better if you define the steps in the README to add it to any project.

```
npm install dex-react
npm install <dependencies like truffle>
...
```

This way is more explicit what we depend on.

WDYT?

Reply:
I disagree.
For every other project, owl-token, solidity-datastructures, etc, we always create the npm modules, such that later on others can easily build on them.
And we benefit from that a lot. For example, imagine the OWL contracts, all libraries, and all the other things are would be in this dex-contracts project. Then it would be harder to navigate for sure.

Reply:
>I expect that many projects won't even do this, and they will just use an interface or the original contract and use the rinkeby/mainnet deployments. Sometimes that approach is more straightforward.

If you wanna write any meaningful tests, then you need to have the deployment scripts etc...

Reply:
I've got a bit lost. 

My understanding is that we want to decide if we add the dependencies for, for example `truffle-contracts` in our dependencies.

I would say to not do this, even as peerDependencies. A simplistic integration might be interested ONLY in the networks.json and the ABI, so we don't want to bring all dependencies.

This is why, I thought of adding additional manual npm install. An alternative would be to create another npm-package or using yarn workspaces to create a package that depends on `dex-contracts` and all the other stuff

Reply:
> If you wanna write any meaningful tests, then you need to have the deployment scripts etc..

I agree, for some people. 
Other people can be happy with `dex-contracts` tests, mock your contracts, and test their interactions with them (mocked)

Reply:
Okay, @anxolin :) then we are on the same page. Sorry, I was also a little bit confused. But actually we were in favor of the same thing!

I will not list all the dependencies in a readme in https://github.com/gnosis/dex-contracts-integration-example, they just go into the package.json of the dex-contracts-integration-example.